### PR TITLE
Fixing up FEATURE_EVENTSOURCE_XPLAT

### PR DESCRIPTION
**Pull request by ramarag**
Fixes https://github.com/dotnet/coreclr/issues/4403

@manu-silicon : Please verify that this does not break your sceanrio
@jkotas  PTAL

### CODE REVIEW

File: src/vm/nativeeventsource.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+//
+// File: nativeeventsource.cpp
+// Abstract: This module implements native part of Event Source support in VM
+//
+
+//
```
Comment (jkotas) [inline src/vm/nativeeventsource.cpp]: Since you have created a dedicated .cpp for this, you should create a dedicated header file as well.

**Comment by jkotas** [comment]
Also, take a look at the build errors...

**Comment by ramarag** [comment]
@jkotas Fixed the errors

**Comment by manu-silicon** [comment]
Ok. Will do on Monday.

File: src/vm/CMakeLists.txt [hunk 2]
```diff
@@ -245,6 +245,7 @@ set(VM_SOURCES_WKS
     typeparse.cpp
     verifier.cpp
     weakreferencenative.cpp
+    nativeeventsource.cpp
```
Comment (jkotas) [inline src/vm/CMakeLists.txt]: The files in this list are in alphabetical order. Could you please add nativeeventsource.cpp to the appropriate spot?

File: src/vm/mscorlib.cpp [hunk 3]
```diff
@@ -132,9 +132,7 @@
 
 #if defined(FEATURE_EVENTSOURCE_XPLAT)
```
Comment (jkotas) [inline src/vm/mscorlib.cpp]: Nit - delete new line

File: src/vm/nativeeventsource.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+// See the LICENSE file in the project root for more information.
+//
+// File: nativeeventsource.h
+// Abstract: This module implements native part of Event Source support in VM
+//
+
+//
+
+//
+// ============================================================================
```
Comment (jkotas) [inline src/vm/nativeeventsource.h]: Could you please the add the ifdef that allows this header file to be included multiple times - like we have in all other header files:

```
#ifndef _NATIVEEVENTSOURCE_H_
#define _NATIVEEVENTSOURCE_H_
```

**Comment by manu-silicon** [comment]
@ramarag It compiles and runs fine on ARM.

**Comment by ramarag** [comment]
@jkotas made the fixes
@dotnet-bot  test  OSX x64 Checked Build and Test

**Comment by jkotas** [comment]
LGTM

**Pull request merged**